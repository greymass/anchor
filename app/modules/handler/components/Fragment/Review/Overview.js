// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Trans, withTranslation } from 'react-i18next';
import compose from 'lodash/fp/compose';
import { Grid, Header, Segment, Tab } from 'semantic-ui-react';

import PromptReviewControls from '../../../components/Review/Controls';
import ErrorMessage from '../../../components/error';
import PromptFragmentPlaceholderTransactionAction from '../../../components/Fragment/Placeholder/Transaction/Action';
import PromptFragmentTransactionAction from '../../../components/Fragment/Transaction/Action';
import PromptFragmentTransactionActionResourceProvider from '../../../components/Fragment/Transaction/Action/ResourceProvider';

class PromptFragmentReviewOverview extends Component<Props> {
  render() {
    const {
      enableWhitelist,
      error,
      globals,
      hasResourceProviderFee,
      loading,
      modifyWhitelist,
      pair,
      t,
      transaction,
      usingResourceProvider,
      whitelist,
    } = this.props;
    const { pricefeed } = globals;
    return (
      <React.Fragment>
        {(loading)
          ? <PromptFragmentPlaceholderTransactionAction />
          : false
        }
        {(transaction && usingResourceProvider)
          ? (
            <PromptFragmentTransactionActionResourceProvider
              enableWhitelist={enableWhitelist}
              hasResourceProviderFee={hasResourceProviderFee}
              modifyWhitelist={modifyWhitelist}
              pair={pair}
              pricefeed={pricefeed}
              total={transaction.actions.length}
              transaction={transaction}
              usingResourceProvider={usingResourceProvider}
              whitelist={whitelist}
            />
          )
          : false
        }
        {(transaction)
          ? transaction.actions.map((action, index) => (
            <PromptFragmentTransactionAction
              action={action}
              enableWhitelist={enableWhitelist}
              modifyWhitelist={modifyWhitelist}
              index={index}
              total={transaction.actions.length}
              whitelist={whitelist}
            />
          ))
          : false
        }
        {(error)
          ? (
            <Segment attached>
              <ErrorMessage
                error={error}
              />
            </Segment>
          )
          : false
        }
      </React.Fragment>
    );
  }
}

export default withTranslation('handler')(PromptFragmentReviewOverview);
