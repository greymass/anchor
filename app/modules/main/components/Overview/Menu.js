// @flow
import React, { Component } from 'react';
import { Menu } from 'semantic-ui-react';
import { withTranslation } from 'react-i18next';

class OverviewMenu extends Component<Props> {
  componentDidUpdate(prevProps, prevState) {
    Object.entries(this.props).forEach(([key, val]) =>
      prevProps[key] !== val && console.log(`OverviewMenu '${key}' changed`)
    );
  }
  render() {
    const {
      stakedResources,
      view,
      viewChange,
      t,
    } = this.props;
    return (
      <Menu pointing secondary size="small">
        <Menu.Item
          active={view === 'systemtokens'}
          content={t('main_components_overview_menu_item_one')}
          icon="cube"
          name="systemtokens"
          onClick={viewChange}
        />
        <Menu.Item
          active={view === 'balances'}
          content={t('main_components_overview_menu_item_two')}
          icon="cubes"
          name="balances"
          onClick={viewChange}
        />
        {(stakedResources)
          ? (
            <Menu.Item
              active={view === 'resources'}
              content={t('main_components_overview_menu_item_three')}
              icon="battery full"
              name="resources"
              onClick={viewChange}
            />
          )
          : false
        }
        <Menu.Item
          active={view === 'governance'}
          content={t('main_components_overview_menu_item_four')}
          icon="gavel"
          name="governance"
          onClick={viewChange}
        />
      </Menu>
    );
  }
}

export default withTranslation('main')(OverviewMenu);
