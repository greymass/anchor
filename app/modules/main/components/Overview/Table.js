// @flow
import React, { Component } from 'react';
import { Header, Table } from 'semantic-ui-react';
import { isEmpty, map, sortBy, uniq } from 'lodash';

import { withTranslation } from 'react-i18next';

import OverviewTableHeader from './Table/Header';

import GlobalAccountFragmentRamUnits from '../../../../shared/containers/Global/Account/Fragment/Ram/Units';
import GlobalAccountFragmentResourceUnits from '../../../../shared/containers/Global/Account/Fragment/Resource/Units';
import GlobalAccountFragmentREXBalance from '../../../../shared/containers/Global/Account/Fragment/REX/Balance';
import GlobalAccountFragmentStaleness from '../../../../shared/containers/Global/Account/Fragment/Staleness';
import GlobalAccountFragmentSystemTokenBalance from '../../../../shared/containers/Global/Account/Fragment/SystemTokenBalance';
import GlobalAccountFragmentSystemTokenValue from '../../../../shared/containers/Global/Account/Fragment/SystemTokenValue';
import GlobalAccountFragmentTokenBalance from '../../../../shared/containers/Global/Account/Fragment/TokenBalance';
import GlobalAccountFragmentTokenDelegated from '../../../../shared/containers/Global/Account/Fragment/TokenDelegated';
import GlobalAccountFragmentTokenRefunding from '../../../../shared/containers/Global/Account/Fragment/TokenRefunding';
import GlobalAccountFragmentTokenStaked from '../../../../shared/containers/Global/Account/Fragment/TokenStaked';
import GlobalAccountFragmentVoterInfoVotes from '../../../../shared/containers/Global/Account/Fragment/VoterInfo/Votes';
import GlobalAccountFragmentVoterInfoEffectiveness from '../../../../shared/containers/Global/Account/Fragment/VoterInfo/Effectiveness';
import GlobalAccountFragmentVoterInfoWeightValue from '../../../../shared/containers/Global/Account/Fragment/VoterInfo/WeightValue';
import GlobalAccountFragmentVoterInfoProxy from '../../../../shared/containers/Global/Account/Fragment/VoterInfo/Proxy';

class OverviewTable extends Component<Props> {
  render() {
    const {
      chainSymbol,
      settings,
      stakedResources,
      supportedContracts,
      t,
      view,
      wallets,
    } = this.props;
    const accountBalances = {};
    const chainAccounts = [].concat(wallets).filter((w) => (w.chainId === settings.chainId));
    const accountNames = uniq(map(chainAccounts, 'account')).sort();

    const tokens = settings.customTokens.filter((token) => {
      const [chain] = token.split(':');
      return (chain === settings.chainId);
    }).sort((a, b) => {
      const [, , asymbol] = a.split(':');
      const [, , bsymbol] = b.split(':');
      return asymbol > bsymbol ? 1 : -1;
    });

    accountNames.forEach((accountName) => {
      tokens.forEach((token) => {
        const [chain, contract, symbol] = token.split(':');
        if (chain === settings.chainId) {
          if (!accountBalances[accountName]) {
            accountBalances[accountName] = [];
          }
          accountBalances[accountName].push((
            <Table.Cell textAlign="right">
              <GlobalAccountFragmentTokenBalance
                account={accountName}
                chainId={settings.chainId}
                contract={contract}
                token={symbol}
              />
            </Table.Cell>
          ));
        }
      });
    });

    return (
      <div style={{ overflowX: 'auto', maxWidth: '75vw' }}>
        <Table celled className="overview" compact size="small" striped unstackable>
          <OverviewTableHeader
            settings={settings}
            stakedResources={stakedResources}
            supportedContracts={supportedContracts}
            tokens={tokens}
            view={view}
          />
          <Table.Body>
            {accountNames.map((accountName) => {
              const addresses = sortBy(
                wallets.filter((wallet) => (wallet.address && wallet.account === accountName)),
                (address) => address.address
              );
              return (
                <Table.Row>
                  <Table.Cell collapsing textAlign="right">
                    <strong>{accountName}</strong>
                    {(addresses.length > 0)
                      ? (
                        <div>
                          {addresses.map((address) => (
                            <div>{address.address}</div>
                          ))}
                        </div>
                      )
                      : false
                    }
                  </Table.Cell>
                  <Table.Cell collapsing>
                    <GlobalAccountFragmentStaleness
                      account={accountName}
                    />
                  </Table.Cell>
                  {(view === 'systemtokens')
                    ? (
                      <React.Fragment>
                        <Table.Cell textAlign="right">
                          <Header size="small">
                            <GlobalAccountFragmentSystemTokenBalance
                              account={accountName}
                              chainId={settings.chainId}
                              contract="eosio"
                              token={chainSymbol}
                            />
                          </Header>
                        </Table.Cell>
                        <Table.Cell textAlign="right">
                          <Header size="small">
                            <GlobalAccountFragmentTokenBalance
                              account={accountName}
                              chainId={settings.chainId}
                              contract="eosio"
                              token={chainSymbol}
                            />
                          </Header>
                        </Table.Cell>
                        {(supportedContracts && supportedContracts.includes('delphioracle'))
                          ? (
                            <Table.Cell textAlign="right">
                              <Header size="small">
                                <GlobalAccountFragmentSystemTokenValue
                                  account={accountName}
                                  chainId={settings.chainId}
                                  contract="eosio"
                                  token={chainSymbol}
                                />
                              </Header>
                            </Table.Cell>
                          )
                          : false
                        }
                        {(stakedResources)
                          ? (
                            <Table.Cell textAlign="right">
                              <GlobalAccountFragmentTokenStaked
                                account={accountName}
                                chainId={settings.chainId}
                                contract="eosio"
                                token={chainSymbol}
                              />
                            </Table.Cell>
                          )
                          : false
                        }
                        {(supportedContracts && supportedContracts.includes('rex'))
                          ? (
                            <Table.Cell textAlign="right">
                              <GlobalAccountFragmentREXBalance
                                account={accountName}
                                chainId={settings.chainId}
                                contract="eosio"
                                token={chainSymbol}
                              />
                            </Table.Cell>
                          )
                          : false
                        }
                        {(stakedResources)
                          ? (
                            <Table.Cell textAlign="right">
                              <GlobalAccountFragmentTokenDelegated
                                account={accountName}
                                chainId={settings.chainId}
                                contract="eosio"
                                token={chainSymbol}
                              />
                            </Table.Cell>
                          )
                          : false
                        }
                        {(stakedResources)
                          ? (
                            <Table.Cell textAlign="right">
                              <GlobalAccountFragmentTokenRefunding
                                account={accountName}
                                chainId={settings.chainId}
                                contract="eosio"
                                token={chainSymbol}
                              />
                            </Table.Cell>
                          )
                          : false
                        }
                      </React.Fragment>
                    )
                    : false
                  }
                  {(view === 'governance')
                    ? (
                      <React.Fragment>
                        <Table.Cell textAlign="right">
                          <GlobalAccountFragmentVoterInfoWeightValue
                            account={accountName}
                          />
                        </Table.Cell>
                        <Table.Cell textAlign="right" collapsing>
                          <GlobalAccountFragmentVoterInfoEffectiveness
                            account={accountName}
                          />
                        </Table.Cell>
                        <Table.Cell textAlign="right" collapsing>
                          <GlobalAccountFragmentVoterInfoVotes
                            account={accountName}
                          />
                        </Table.Cell>
                        <Table.Cell>
                          <GlobalAccountFragmentVoterInfoProxy
                            account={accountName}
                          />
                        </Table.Cell>
                      </React.Fragment>
                    )
                    : false
                  }
                  {(view === 'balances')
                    ? (
                      <React.Fragment>
                        {(isEmpty(accountBalances))
                          ? (
                            <Table.Cell>
                              {t('main_components_overview_no_token_balances')}
                            </Table.Cell>
                          )
                          : accountBalances[accountName]
                        }
                      </React.Fragment>
                    )
                    : false
                  }
                  {(view === 'resources')
                    ? (
                      <React.Fragment>
                        <Table.Cell textAlign="right">
                          <GlobalAccountFragmentResourceUnits
                            account={accountName}
                            type="cpu"
                          />
                        </Table.Cell>
                        <Table.Cell textAlign="right">
                          <GlobalAccountFragmentResourceUnits
                            account={accountName}
                            type="net"
                          />
                        </Table.Cell>
                        <Table.Cell textAlign="right">
                          <GlobalAccountFragmentRamUnits
                            account={accountName}
                          />
                        </Table.Cell>
                      </React.Fragment>
                    )
                    : false
                  }
                </Table.Row>
              );
            })}
          </Table.Body>
        </Table>
      </div>
    );
  }
}

export default withTranslation('main')(OverviewTable);
