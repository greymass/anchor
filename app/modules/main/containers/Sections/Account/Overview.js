// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withTranslation } from 'react-i18next';
import compose from 'lodash/fp/compose';
import { withRouter } from 'react-router-dom';
import { Button, Container, Header, Icon, Segment } from 'semantic-ui-react';
import { Resources } from '@greymass/eosio-resources';
import AccountHeader from './Header';
import AccountOverviewRam from './Overview/Ram';
import AccountOverviewResource from './Overview/Resource';

import * as AccountsActions from '../../../../../shared/actions/accounts';
import * as NavigationActions from '../../../actions/navigation';

class AccountOverview extends Component<Props> {
  constructor(props) {
    super(props);
    const account = props.match.params.account_name;
    const loaded = (
      Object.keys(this.props.accounts).includes(account)
      && Object.keys(this.props.balances).includes(account)
    );
    this.state = {
      account,
      expanded: {},
      loaded,
      sample: false,
      rstate: false,
      pstate: false,
    };
  }
  componentDidMount() {
    const { settings } = this.props;
    const { account } = settings;
    if (
      // Missing account
      !Object.keys(this.props.accounts).includes(account)
      // Missing balance
      || !Object.keys(this.props.balances).includes(account)
    ) {
      this.props.actions.getAccount(account);
    }
    this.initProviders();
  }
  componentDidUpdate(prevProps) {
    const { connection, settings } = this.props;
    // If the user changes the current account, change the UI
    if (connection.httpEndpoint !== prevProps.connection.httpEndpoint) {
      this.initProviders();
    }
    if (
      settings.account !== prevProps.settings.account
      || settings.chainId !== prevProps.settings.chainId
    ) {
      this.setState({
        account: settings.account,
        loaded: false,
      }, () => this.props.actions.changeModule(`account/${settings.account}`));
    }
    // Let the UI know when the account is loaded
    const { loaded } = this.state;
    if (!loaded) {
      const { accounts, balances } = this.props;
      if (
        Object.keys(accounts).includes(settings.account)
        && Object.keys(balances).includes(settings.account)
      ) {
        this.setState({ loaded: true });
      }
    }
  }
  initProviders = async () => {
    const { connection } = this.props;
    const features = [];
    if (connection.supportedContracts.includes('rex')) {
      features.push('rex');
    }
    if (connection.supportedContracts.includes('powerup')) {
      features.push('powerup');
    }
    this.resources = new Resources({ url: connection.httpEndpoint });
    this.setState({
      sample: false,
      pstate: false,
      rstate: false,
    }, async () => {
      const rexEnabled = features.includes('rex');
      const powerupEnabled = features.includes('powerup');
      if (rexEnabled || powerupEnabled) {
        const sample = await this.resources.getSampledUsage();
        this.setState({ sample });
      }
      if (rexEnabled) {
        const rstate = await this.resources.v1.rex.get_state();
        this.setState({ rstate });
      }
      if (powerupEnabled) {
        const pstate = await this.resources.v1.powerup.get_state();
        this.setState({ pstate });
      }
    });
  }
  toggleExpand = (e, { name }) => {
    this.setState({
      expanded: Object.assign({}, this.state.expanded, {
        [name]: !this.state.expanded[name]
      })
    });
  }
  refresh = () => {
    const { getAccount } = this.props.actions;
    getAccount(this.state.account);
  }
  render() {
    const {
      system,
      t,
    } = this.props;
    const {
      account,
      loaded,
      sample,
      pstate,
      rstate,
    } = this.state;
    if (!account || account === 'undefined') {
      return (
        <Segment>
          <Header>
            {t('main_sections_overview_header_two')}
            <Header.Subheader>
              {t('main_sections_overview_subheader_two')}
            </Header.Subheader>
          </Header>
        </Segment>
      );
    }
    const refreshing = (system.GET_ACCOUNT_LAST_ACCOUNT === account && system.GET_ACCOUNT === 'PENDING');
    return (
      <React.Fragment>
        <Container fluid clearing>
          <Button
            basic
            color="grey"
            content={t('main_sections_overview_button_one')}
            floated="right"
            icon="refresh"
            loading={refreshing}
            onClick={this.refresh}
          />
          <Header
            content={t('main_sections_overview_header_one')}
            subheader={t('main_sections_overview_subheader_one')}
            style={{ margin: 0 }}
          />
        </Container>
        {(loaded)
          ? (
            <React.Fragment>
              <AccountHeader
                account={account}
              />
              <AccountOverviewRam
                account={account}
              />
              <AccountOverviewResource
                account={account}
                pstate={pstate}
                sample={sample}
                resource="cpu"
                rstate={rstate}
              />
              <AccountOverviewResource
                account={account}
                pstate={pstate}
                sample={sample}
                resource="net"
                rstate={rstate}
              />
            </React.Fragment>
          )
          : false
        }
        {(!loaded)
          ? (
            <Segment>
              <Header icon size="large" textAlign="center">
                <Icon
                  loading
                  name="circle notched"
                />
                {t('main_sections_overview_header_two')}
                <Header.Subheader
                  content={t('main_sections_overview_subheader_two')}
                />
              </Header>
            </Segment>
          )
          : false
        }
      </React.Fragment>
    );
  }
}

function mapStateToProps(state) {
  return {
    accounts: state.accounts,
    balances: state.balances,
    connection: state.connection,
    resources: state.resources,
    settings: state.settings,
    system: state.system,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      ...NavigationActions,
      ...AccountsActions,
    }, dispatch)
  };
}

export default compose(
  withRouter,
  withTranslation('main'),
  connect(mapStateToProps, mapDispatchToProps)
)(AccountOverview);
