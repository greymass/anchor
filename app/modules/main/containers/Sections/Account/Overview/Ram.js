// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Trans, withTranslation } from 'react-i18next';
import compose from 'lodash/fp/compose';

import { Button, Container, Grid, Header, Progress, Segment, Table } from 'semantic-ui-react';

import AccountOverviewRamBuy from './Ram/Buy';
import AccountOverviewRamSell from './Ram/Sell';

import GlobalAccountFragmentRamUnits from '../../../../../../shared/containers/Global/Account/Fragment/Ram/Units';
import GlobalAccountFragmentRamPrice from '../../../../../../shared/containers/Global/Account/Fragment/Ram/Price';
import GlobalAccountFragmentRamProgress from '../../../../../../shared/containers/Global/Account/Fragment/Ram/Progress';
import GlobalAccountFragmentRamUsage from '../../../../../../shared/containers/Global/Account/Fragment/Ram/Usage';
import GlobalAccountFragmentRamMax from '../../../../../../shared/containers/Global/Account/Fragment/Ram/Max';
import GlobalAccountFragmentRamValue from '../../../../../../shared/containers/Global/Account/Fragment/Ram/Value';

class AccountOverviewRam extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false,
    };
  }
  toggleExpand = () => this.setState({
    expanded: !this.state.expanded
  });
  render() {
    const {
      account,
      connection,
      settings,
      t,
    } = this.props;
    const resource = 'ram';
    const {
      expanded,
    } = this.state;
    return (
      <Segment>
        <Grid divided fluid stackable>
          <Grid.Row>
            <Grid.Column width={4}>
              <Header size="large" textAlign="center">
                <Header.Content>
                  RAM
                  <Header.Subheader
                    style={{
                      marginTop: '0.35em',
                    }}
                  >
                    <Trans i18nKey="main_sections_overview_ram_subheader" t={t}>
                      A <strong>size-based</strong> resource an account can use to <strong>store data</strong> within smart contracts.
                    </Trans>
                  </Header.Subheader>
                </Header.Content>
              </Header>
            </Grid.Column>
            <Grid.Column width={12}>
              <Grid stackable>
                <Grid.Row>
                  <Grid.Column width={7}>
                    <Header textAlign="center">
                      <GlobalAccountFragmentRamUnits
                        account={account}
                      />
                      <Header.Subheader>
                        {(settings.displayResourcesAvailable)
                          ? t(
                            'main_sections_overview_resource_grid_subheader_three_available',
                            { resource: resource.toUpperCase() }
                          ) : t(
                            'main_sections_overview_resource_grid_subheader_three_used',
                            { resource: resource.toUpperCase() }
                          )
                        }
                      </Header.Subheader>
                    </Header>
                    <Segment basic style={{ padding: 0 }}>
                      <GlobalAccountFragmentRamProgress
                        account={account}
                        size="tiny"
                        style={{ marginBottom: 0 }}
                        type={resource}
                      />
                      <Table
                        definition
                        size="small"
                        style={{
                          display: (expanded ? 'table' : 'none'),
                          marginBottom: 0,
                        }}
                        unstackable
                      >
                        <Table.Row>
                          <Table.Cell collapsing>{t('main_sections_overview_resource_used')}</Table.Cell>
                          <Table.Cell>
                            <GlobalAccountFragmentRamUsage
                              account={account}
                            />
                          </Table.Cell>
                        </Table.Row>
                        <Table.Row>
                          <Table.Cell collapsing>{t('main_sections_overview_resource_allowed')}</Table.Cell>
                          <Table.Cell>
                            <GlobalAccountFragmentRamMax
                              account={account}
                              type={resource}
                            />
                          </Table.Cell>
                        </Table.Row>
                      </Table>
                    </Segment>
                  </Grid.Column>
                  <Grid.Column
                    textAlign="center"
                    verticalAlign="top"
                    width={2}
                  >
                    <Button
                      basic
                      icon={`angle double ${(expanded ? 'up' : 'down')}`}
                      name={resource}
                      onClick={this.toggleExpand}
                      size="small"
                      style={{
                        marginTop: '5.25em'
                      }}
                    />
                  </Grid.Column>
                  <Grid.Column width={7}>
                    <Header textAlign="center">
                      <GlobalAccountFragmentRamMax
                        account={account}
                      />
                      <Header.Subheader>
                        {t('main_sections_overview_ram_grid_subheader')}
                      </Header.Subheader>
                    </Header>
                    <Progress
                      percent={100}
                      size="tiny"
                      style={{ marginBottom: 0 }}
                    />
                    <Table
                      definition
                      size="small"
                      style={{
                        display: (expanded ? 'table' : 'none'),
                        marginBottom: 0,
                      }}
                      unstackable
                    >
                      <Table.Row>
                        <Table.Cell collapsing>{t('main_sections_overview_ram_table_cell_one')}</Table.Cell>
                        <Table.Cell>
                          <GlobalAccountFragmentRamValue
                            account={account}
                            precision={4}
                          />
                          {` ${connection.chainRamSymbol || connection.chainSymbol}`}
                        </Table.Cell>
                      </Table.Row>
                      <Table.Row>
                        <Table.Cell collapsing>{t('main_sections_overview_ram_table_cell_two')}</Table.Cell>
                        <Table.Cell>
                          <GlobalAccountFragmentRamPrice
                            precision={8}
                          />
                          {` ${connection.chainRamSymbol || connection.chainSymbol}`}/byte
                        </Table.Cell>
                      </Table.Row>
                    </Table>
                    <Container
                      fluid
                      style={{
                        marginTop: '1em'
                      }}
                    >
                      <AccountOverviewRamBuy
                        account={account}
                      />
                      <AccountOverviewRamSell
                        account={account}
                      />
                    </Container>
                  </Grid.Column>
                </Grid.Row>
              </Grid>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Segment>
    );
  }
}

function mapStateToProps(state) {
  return {
    connection: state.connection,
    settings: state.settings,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({

    }, dispatch)
  };
}

export default compose(
  withTranslation('main'),
  withRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(AccountOverviewRam);
