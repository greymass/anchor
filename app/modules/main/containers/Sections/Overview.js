// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Button, Container, Grid, Header, Icon, Label, Segment } from 'semantic-ui-react';

import { withTranslation } from 'react-i18next';
import compose from 'lodash/fp/compose';
import { uniq } from 'lodash';

import OverviewSidebarContainer from './Overview/Sidebar';

import { getAccount } from '../../../../shared/actions/accounts';
import NavigationActions from '../../actions/navigation';
import OverviewMenu from '../../components/Overview/Menu';
import OverviewTable from '../../components/Overview/Table';

class OverviewContainer extends Component<Props> {
  state = {
    view: 'systemtokens'
  };
  viewChange = (e, data) => this.setState({ view: data.name })
  refreshAccounts = (accountOnly = true) => {
    const { settings, wallets } = this.props;
    const accounts = uniq(wallets.filter((wallet) => wallet.chainId === settings.chainId).map((wallet) => wallet.account));
    accounts.forEach((account) => this.props.actions.getAccount(account, accountOnly));
  }
  render() {
    const {
      chainSymbol,
      pricefeed,
      settings,
      stakedResources,
      supportedContracts,
      t,
      wallets,
    } = this.props;
    const {
      view
    } = this.state;
    if (!settings.walletInit) {
      return false;
    }
    let header;
    switch (view) {
      default:
      case 'systemtokens': {
        header = (
          <Header>
            {t('main_sections_overview_system_tokens_header', { chainSymbol })}

            <Header.Subheader>
              {t('main_sections_overview_system_tokens_subheader')}
            </Header.Subheader>
          </Header>
        );
        break;
      }
      case 'balances': {
        header = (
          <Header>
            {t('main_sections_overview_balances_header')}
            <Header.Subheader>
              {t('main_sections_overview_balances_subheader')}
            </Header.Subheader>
          </Header>
        );
        break;
      }
      case 'governance': {
        header = (
          <Header>
            {t('main_sections_overview_governance_header')}
            <Header.Subheader>
              {t('main_sections_overview_governance_subheader')}
            </Header.Subheader>
          </Header>
        );
        break;
      }
      case 'resources': {
        header = (
          <Header>
            {t('main_sections_overview_resources_header')}
            <Header.Subheader>
              {t('main_sections_overview_resources_subheader')}
            </Header.Subheader>
          </Header>
        );
        break;
      }
    }
    let pair = 'eosusd';
    switch (settings.chainId) {
      case '1064487b3cd1a897ce03ae5b6a865651747e2e152090f99c1d19d44e01aea5a4': {
        pair = 'waxpusd';
        break;
      }
      default: {
        break;
      }
    }
    return (
      <React.Fragment>
        <Grid stackable>
          <Grid.Row>
            <Grid.Column width={12}>
              <Header
                content={t('main_sections_overview_grid_header')}
                subheader={t('main_sections_overview_grid_subheader')}
              />
              <OverviewMenu
                stakedResources={stakedResources}
                view={view}
                viewChange={this.viewChange}
              />
              <Segment color="green" style={{ marginTop: 0 }}>
                <Grid style={{ marginBottom: 0 }}>
                  <Grid.Row>
                    <Grid.Column width={13}>
                      {header}
                    </Grid.Column>
                    <Grid.Column textAlign="right" width={3}>
                      {(['systemtokens'].includes(view) && (supportedContracts && supportedContracts.includes('delphioracle')))
                        ? (
                          <Label
                            color="green"
                            content={(
                              <span style={{
                                display: 'block',
                                textAlign: 'center',
                              }}
                              >
                                {(pricefeed && pricefeed[pair])
                                  ? (
                                    <span>${(pricefeed[pair] / 10000).toFixed(2)}</span>
                                  )
                                  : (
                                    <Icon name="clock" />
                                  )
                                }

                                {' '}
                                <span style={{
                                  display: 'block',
                                  marginTop: '0.25em',
                                  fontSize: '0.70em',
                                }}
                                >
                                  USD/{chainSymbol}
                                </span>

                              </span>
                            )}
                          />

                        )
                        : false
                      }
                    </Grid.Column>
                  </Grid.Row>
                </Grid>
                <OverviewTable
                  chainSymbol={chainSymbol}
                  pricefeed={pricefeed}
                  settings={settings}
                  supportedContracts={supportedContracts}
                  stakedResources={stakedResources}
                  view={view}
                  wallets={wallets}
                />
                <Container
                  fluid
                  style={{ marginTop: '1em' }}
                  textAlign="center"
                >

                  {(view === 'balances')
                    ? (
                      <React.Fragment>
                        <Button
                          content={t('main_sections_wallet_transfer_button_refresh')}
                          icon="cubes"
                          onClick={() => this.refreshAccounts(false)}
                          primary
                        />
                        <Button
                          basic
                          content={t('main_sections_overview_container_button_tokens')}
                          icon="cubes"
                          onClick={() => this.props.actions.changeModule('tools/customtokens')}
                          primary
                        />
                      </React.Fragment>
                    )
                    : (
                      <React.Fragment>
                        <Button
                          content={t('main_sections_overview_container_button_accounts')}
                          icon="users"
                          onClick={() => this.refreshAccounts(true)}
                          primary
                        />
                        <Button
                          basic
                          content={t('main_sections_overview_container_button')}
                          icon="users"
                          onClick={() => this.props.actions.changeModule('manage/wallets')}
                          primary
                        />
                      </React.Fragment>
                    )
                  }
                </Container>
              </Segment>
            </Grid.Column>
            <Grid.Column width={4}>
              <OverviewSidebarContainer />
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </React.Fragment>
    );
  }
}

function mapStateToProps(state) {
  return {
    balances: state.balances,
    chainSymbol: state.connection.chainSymbol,
    connection: state.connection,
    pricefeed: state.globals.pricefeed,
    navigation: state.navigation,
    settings: state.settings,
    stakedResources: state.connection.stakedResources,
    supportedContracts: state.connection.supportedContracts,
    wallets: state.wallets
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      getAccount,
      ...NavigationActions,
    }, dispatch)
  };
}

export default compose(
  withTranslation('main'),
  connect(mapStateToProps, mapDispatchToProps)
)(OverviewContainer);
