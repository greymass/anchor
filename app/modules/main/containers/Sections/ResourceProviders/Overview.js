// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Segment } from 'semantic-ui-react';

class ResourceProvidersOverview extends Component<Props> {
  render = () => (
    <Segment>
      Resource Provider Overview
    </Segment>
  )
}

function mapStateToProps(state) {
  return {

  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({

    }, dispatch)
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ResourceProvidersOverview));
