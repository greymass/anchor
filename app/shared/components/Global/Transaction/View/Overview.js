// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Statistic } from 'semantic-ui-react';

class GlobalTransactionViewDetail extends Component<Props> {
  render() {
    const {
      t,
      transaction
    } = this.props;
    const {
      data,
    } = transaction;
    return (
      <Statistic.Group>
        <Statistic>
          <Statistic.Value>
            {data.transaction.transaction.actions.length}
          </Statistic.Value>
          <Statistic.Label>Contract(s)</Statistic.Label>
        </Statistic>
        <Statistic>
          <Statistic.Value>
            {data.transaction.transaction.actions.length}
          </Statistic.Value>
          <Statistic.Label>Action(s)</Statistic.Label>
        </Statistic>
      </Statistic.Group>
    );
  }
}

export default translate('global')(GlobalTransactionViewDetail);
