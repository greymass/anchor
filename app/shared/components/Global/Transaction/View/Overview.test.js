import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalTransactionViewDetail } from './Overview';

const mockProps = {
  t: text => text,
  transaction: {
    data: {
      transaction: {
        transaction: {
          actions: []
        }
      }
    }
  },
};

describe('GlobalTransactionViewOverview', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalTransactionViewDetail {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

