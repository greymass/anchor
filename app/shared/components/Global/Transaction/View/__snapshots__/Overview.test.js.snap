// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTransactionViewOverview renders correctly 1`] = `
<StatisticGroup>
  <Statistic>
    <StatisticValue>
      0
    </StatisticValue>
    <StatisticLabel>
      Contract(s)
    </StatisticLabel>
  </Statistic>
  <Statistic>
    <StatisticValue>
      0
    </StatisticValue>
    <StatisticLabel>
      Action(s)
    </StatisticLabel>
  </Statistic>
</StatisticGroup>
`;
